### ISSUE TITLE
testing installed files?

### ISSUE THREAD

**Issue by dschwoerer** [boutproject/xBOUT#119]
Is it possible to test that the installed files are working as expected?

I tried to run `pytest` with the installed files in `PYTHONPATH` - but that fails:
```
collected 9 items / 7 errors / 1 skipped / 1 selected

==================================== ERRORS ====================================
_________________ ERROR collecting xbout/tests/test_animate.py _________________
import file mismatch:
imported module 'xbout.tests.test_animate' has this __file__ attribute:
  /builddir/build/BUILDROOT/python-xBOUT-0.0.1-1.fc33.x86_64/usr/lib/python3.8/site-packages/xbout/tests/test_animate.py
which is not the same as the test file we want to collect:
  /builddir/build/BUILD/xBOUT-master/xbout/tests/test_animate.py
HINT: remove __pycache__ / .pyc files and/or use a unique basename for your test file modules
```

If that is not possible, that is fine as well, but I would prefer testing the installed files, rather the source.

It shouldn't be a huge issue, so 'Won't fix' is fine to.

There are also some warnings:
```
/usr/lib/python3.8/site-packages/xbout/load.py:171: SyntaxWarning: "is" with a literal. Did you mean "=="?
/usr/lib/python3.8/site-packages/xbout/load.py:556: SyntaxWarning: "is" with a literal. Did you mean "=="?
/usr/lib/python3.8/site-packages/xbout/load.py:558: SyntaxWarning: "is" with a literal. Did you mean "=="?
/usr/lib/python3.8/site-packages/xbout/tests/test_load.py:290: SyntaxWarning: "is" with a literal. Did you mean "=="?
/usr/lib/python3.8/site-packages/xbout/tests/test_load.py:294: SyntaxWarning: "is" with a literal. Did you mean "=="?
/usr/lib/python3.8/site-packages/xbout/tests/test_load.py:311: SyntaxWarning: "is" with a literal. Did you mean "=="?
/usr/lib/python3.8/site-packages/xbout/load.py:171: SyntaxWarning: "is" with a literal. Did you mean "=="?
/usr/lib/python3.8/site-packages/xbout/load.py:556: SyntaxWarning: "is" with a literal. Did you mean "=="?
/usr/lib/python3.8/site-packages/xbout/load.py:558: SyntaxWarning: "is" with a literal. Did you mean "=="?
/usr/lib/python3.8/site-packages/xbout/tests/test_load.py:290: SyntaxWarning: "is" with a literal. Did you mean "=="?
/usr/lib/python3.8/site-packages/xbout/tests/test_load.py:294: SyntaxWarning: "is" with a literal. Did you mean "=="?
/usr/lib/python3.8/site-packages/xbout/tests/test_load.py:311: SyntaxWarning: "is" with a literal. Did you mean "=="?
```

Full output is [here](https://download.copr.fedorainfracloud.org/results/davidsch/testing/fedora-rawhide-x86_64/01344976-python-xBOUT/builder-live.log.gz)

**Comment by johnomotani** [comment]
See here https://docs.pytest.org/en/latest/goodpractices.html#tests-outside-application-code and the next section. I haven't tried installing `xBOUT` - could you try (from that link) `pytest --pyargs xBOUT`?

**Comment by dschwoerer** [comment]
Thanks, that works.

However, I think it only works because the tests are installed with the other files. Not sure that is intended?

Further, `xBOUT` is actually called `xbout`, so it needs to be `pytest --pyargs xbout`.

Should I close this issue, or keep it open as a reminder for the warnings?

**Comment by johnomotani** [comment]
Tests being installed with the other files is one of the approaches recommended by `pytest`, so I think it's intended. Probably worth adding a note about using `pytest --pyargs xbout` to test an installed version to the docs somewhere, then close this.

**Comment by TomNicholas** [comment]
We should fix those `is` warnings too though, it only requires changing `is` -> `==`.

**Comment by johnomotani** [comment]
`pytest --pyargs xbout` to test installed versions is now noted in the documentation.

`is` warnings were fixed in #120.
